apiClient/client.ts

```diff
@@ -232,15 +232,24 @@ export async function login(email: string): Promise<any> {
 }
 
 export async function getUserDetails(
-  token: string
+  token?: string
 ): Promise<ApiUserMetadata | ApiError | LocalError> {
   try {
-    const response = await fetch(`${API_URL}/me`, {
-      headers: {
-        'Content-Type': 'application/json',
-        Authorization: `Bearer ${token}`,
-      },
-    })
+    let response
+    if (!token) {
+      response = await fetch(`${API_URL}/me`, {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      })
+    } else {
+      response = await fetch(`${API_URL}/me`, {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+      })
+    }
```

Comment:
Suggestion: You can do this to avoid duplicating the fetch call:

```ts
  const headers: HeadersInit = {
    'Content-Type': 'application/json',
  }

  if (token) {
    headers.Authorization = `Bearer ${token}`
  }

  const response = await fetch(`${API_URL}/me`, {
    headers,
  })
```